src/Microsoft.Diagnostics.Runtime/ClrValue.cs

```diff
@@ -0,0 +1,1077 @@
+                throw new MemoryReadException(Address);
+
+            return result;
+        }
+
+
+
+        /// <summary>
+        /// Converts the value to a long.  Note that this method is a conversion, not a cast.  Similarly sized data will be converted.
+        /// </summary>
+        /// <returns>This value as a long.</returns>
+        public virtual long AsInt64()
+        {
+            if (ElementType != ClrElementType.UInt64 && ElementType != ClrElementType.Int64)
+                throw new InvalidOperationException("Value is not a long.");
+
+            long result;
+            if (!_runtime.ReadQword(Address, out result))
+                throw new MemoryReadException(Address);
+
+            return result;
+        }
+
+
+        /// <summary>
+        /// Converts the value to a float.
+        /// </summary>
+        /// <returns>This value as a float.</returns>
+        public virtual float AsFloat()
+        {
+            if (ElementType != ClrElementType.Float)
+                throw new InvalidOperationException("Value is not a float.");
+
+            float result;
+            if (!_runtime.ReadFloat(Address, out result))
+                throw new MemoryReadException(Address);
+
+            return result;
+        }
+
+
+
+        /// <summary>
+        /// Converts the value to a double.
+        /// </summary>
+        /// <returns>This value as a double.</returns>
+        public virtual double AsDouble()
+        {
+            if (ElementType != ClrElementType.Double)
+                throw new InvalidOperationException("Value is not a double.");
+
+            double result;
+            if (!_runtime.ReadFloat(Address, out result))
+                throw new MemoryReadException(Address);
+
+            return result;
+        }
+
+
+        /// <summary>
+        /// Converts the value to a raw string.  This method will throw an InvalidOperationException if the target value is not a string.
+        /// </summary>
+        /// <returns>The string contents of this value.</returns>
+        public virtual string AsString()
+        {
+            if (ElementType != ClrElementType.String && (!ClrRuntime.IsObjectReference(ElementType) || !Type.IsString))
+                throw new InvalidOperationException("Value is not a string.");
+
+            ulong str;
+            if (!_runtime.ReadPointer(Address, out str))
+                throw new MemoryReadException(Address);
+
+            if (str == 0)
+                return null;
+
+            string result;
+            if (!_runtime.ReadString(str, out result))
+                throw new MemoryReadException(str);
+
+            return result;
+        }
+
+
+        /// <summary>
+        /// Converts the value to an IntPtr.  Note that this method is a conversion, not a cast.  Similarly sized data will be converted.
+        /// </summary>
+        /// <returns>This value as an IntPtr.</returns>
+        public virtual IntPtr AsIntPtr()
+        {
```

Comment:
This makes the assumption that target pointer and host pointer have the same size. Its always true today for DAC, but I don't know if you want to encourage people to lock themselves into that? I'd encourage people to map IntPtr as a long or ulong so that things keep working if we enable debugging 64 bit processes with 32 bit debuggers